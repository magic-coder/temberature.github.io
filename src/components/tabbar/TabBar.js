import React from "react";
import "./TabBar.less";
import { NavLink } from "react-router-dom";
import OAIcon from "../icon/Icon";
class OATabBar extends React.Component {
  render() {
    return (
      <div className="oa-tabbar is-fixed">
        <NavLink exact to="/" className="oa-tab-item">
          <div className="oa-tab-item-icon">
            <OAIcon type={this.props.icons[0]} />
          </div>
          <div className="oa-tab-item-label">{this.props.titles[0]}</div>
        </NavLink>
        <NavLink to="/profile" className="oa-tab-item">
          <div className="oa-tab-item-icon">
            <OAIcon type={this.props.icons[1]} />
          </div>
          <div className="oa-tab-item-label">{this.props.titles[1]}</div>
        </NavLink>
      </div>
    );
  }
}

export default OATabBar;
