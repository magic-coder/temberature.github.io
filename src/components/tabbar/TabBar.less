.mint-tabbar {
  background-image: -webkit-gradient(
    linear,
    left top,
    left bottom,
    from(#d9d9d9),
    color-stop(50%, #d9d9d9),
    color-stop(50%, transparent)
  );
  background-image: linear-gradient(
    180deg,
    #d9d9d9,
    #d9d9d9 50%,
    transparent 50%
  );
  background-size: 100% 1px;
  background-repeat: no-repeat;
  background-position: top left;
  position: relative;
  background-color: #fafafa;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  right: 0;
  bottom: 0;
  left: 0;
  position: absolute;
  text-align: center;
  .mint-tab-item {
    .am-icon {
      color: #C0CBDE;
    }
    &.is-selected {
      .am-icon {
        color: #0B0B0B;
      }
    }
  }

}
.mint-tabbar > .mint-tab-item.is-selected {
  background-color: #eaeaea;
  color: #26a2ff;
}
.mint-tabbar.is-fixed {
  right: 0;
  bottom: 0;
  left: 0;
  position: fixed;
  z-index: 1;
}
.mint-tab-item {
  display: block;
  padding: 7px 0;
  -webkit-box-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-decoration: none;
}
.mint-tab-item-icon {
  width: 24px;
  height: 24px;
  margin: 0 auto 5px;
}
.mint-tab-item-icon:empty {
  display: none;
}
.mint-tab-item-label {
  color: inherit;
  font-size: 12px;
  line-height: 1;
}
.mint-tab-container-item {
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 100%;
}
.mint-tab-container {
  overflow: hidden;
  position: relative;
}
.mint-tab-container .swipe-transition {
  -webkit-transition: -webkit-transform 150ms ease-in-out;
  transition: -webkit-transform 150ms ease-in-out;
  transition: transform 150ms ease-in-out;
  transition: transform 150ms ease-in-out, -webkit-transform 150ms ease-in-out;
}
.mint-tab-container-wrap {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
}
.mint-tabbar .mint-tab-item {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-top: 1px solid #cacaca;
  padding: 16px 0 17px;
  color: #c0cbde;
  background: #fff;
}
.mint-tabbar .mint-tab-item .mint-tab-item-icon {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  width: initial;
  height: initial;
  margin: 0 4px 0 0;
}
.mint-tabbar .mint-tab-item .mint-tab-item-icon img {
  width: initial;
  height: 16px;
}
.mint-tabbar .mint-tab-item .mint-tab-item-label {
  font-size: 16px;
}
.mint-tabbar .mint-tab-item:first-child {
  border-right: 1px solid #cacaca;
}
.mint-tabbar .mint-tab-item.is-selected {
  color: #0b0b0b;
  background: #fff;
}
.am-icon-md {
  width: 36/2px;
  height: 32/2px;
}
