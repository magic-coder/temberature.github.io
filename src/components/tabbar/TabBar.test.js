import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router, Route } from "react-router-dom";
import TabBar from "./TabBar";

it("renders correctly", function() {
  const tree = renderer
    .create(
      <Router>
        <Route>
          <TabBar
            titles={["新生学院", "个人中心"]}
            icons={["./assets/icon_home.svg", "./assets/icon_profile.svg"]}
          />
        </Route>
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
