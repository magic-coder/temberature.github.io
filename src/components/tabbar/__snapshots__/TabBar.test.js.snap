// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="oa-tabbar is-fixed"
>
  <a
    aria-current="true"
    className="oa-tab-item active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    <div
      className="oa-tab-item-icon"
    >
      <svg
        className="am-icon am-icon-/assets/icon_home.svg am-icon-md "
      >
        <use
          xlinkHref="./assets/icon_home.svg"
        />
         
         
      </svg>
    </div>
    <div
      className="oa-tab-item-label"
    >
      新生学院
    </div>
  </a>
  <a
    aria-current={false}
    className="oa-tab-item"
    href="/profile"
    onClick={[Function]}
    style={undefined}
  >
    <div
      className="oa-tab-item-icon"
    >
      <svg
        className="am-icon am-icon-/assets/icon_profile.svg am-icon-md "
      >
        <use
          xlinkHref="./assets/icon_profile.svg"
        />
         
         
      </svg>
    </div>
    <div
      className="oa-tab-item-label"
    >
      个人中心
    </div>
  </a>
</div>
`;
